src/EnergyPlus/WaterThermalTanks.cc

```diff
@@ -10639,6 +10639,96 @@ namespace WaterThermalTanks {
         return PLRResidualHPWH;
     }
 
+    bool SourceHeatNeed(int const WaterThermalTankNum,
```

Comment:
A nice clean up here would be to either:
- Make this function a method on the tank data structure, that way you could just call `WaterThermalTank(WaterThermalTankNum).sourceHeatNeed(outletTemp, dbTemp, spTemp)` and eliminate the array lookups, or
- Pass in a const tank object as an argument instead of the tank num, again eliminating the array lookups.

Reply:
Thanks, Edwin, I passed the tank object as SourceHeatNeed function argument (option 2)